devito/types/basic.py

```diff
@@ -933,6 +748,7 @@ def __new__(cls, *args, **kwargs):
         # when executing __init_finalize__
         newobj._name = name
         newobj._dimensions = dimensions
+        newobj._staggered = staggered
```

Comment:
so Arrays now get this attribute too, which is not ideal. We may need a further AbstractFunction subclass instead, or something else

Reply:
why did you lift it up here? as in, why not enough in `DiscreteFunction`?

Reply:
Because I need it to be in `__new__`, lifting it up requires a lot of messy tinkering with the init_finalize. But I will have another look.